File: src/main/java/hiiragi283/material/api/tile/HiiragiProvider.java
package hiiragi283.material.api.tile;

import hiiragi283.material.api.capability.HiiragiCapabilityProvider;
import net.minecraftforge.energy.IEnergyStorage;
import net.minecraftforge.fluids.capability.IFluidHandler;
import net.minecraftforge.items.IItemHandler;
import org.jetbrains.annotations.NotNull;

public abstract class HiiragiProvider {

    public interface Inventory {

        @NotNull
        HiiragiCapabilityProvider<IItemHandler> createInventory();

    }

    public interface Tank {

        @NotNull
        HiiragiCapabilityProvider<IFluidHandler> createTank();

    }

    public interface Battery {

        @NotNull
        HiiragiCapabilityProvider<IEnergyStorage> createBattery();

    }

}

File: src/main/kotlin/hiiragi283/material/api/tile/HiiragiProvider.kt
package hiiragi283.material.api.tile

import hiiragi283.material.api.capability.HiiragiCapabilityProvider
import net.minecraftforge.energy.IEnergyStorage
import net.minecraftforge.fluids.capability.IFluidHandler
import net.minecraftforge.items.IItemHandler


object HiiragiProvider {

    interface Inventory {

        fun createInventory(): HiiragiCapabilityProvider<IItemHandler>

    }

    interface Tank {

        fun createTank(): HiiragiCapabilityProvider<IFluidHandler>

    }

    interface Energy {

        fun createBattery(): HiiragiCapabilityProvider<IEnergyStorage>

    }

}